pygmt/session_management.py

```diff
@@ -12,6 +16,10 @@ def begin():
 
     Only meant to be used once for creating the global session.
     """
+    # On Windows, need to set GMT_SESSION_NAME to a unique value
+    if sys.platform == "win32":
+        os.environ["GMT_SESSION_NAME"] = unique_name()
```

Comment:
Hmm, I thought this should best be handled upstream in GMT, but then I saw the note from https://github.com/GenericMappingTools/gmt/blob/104bcdbafd13eb96260b445910ce3587a362dbc0/src/gmt_api.c#L1050-L1057:

```
	/* OK, the trouble is the following. On Win, if the Windows executables are run from within a bash window
	   gmtapi_get_ppid returns different values for each call, and this completely breaks the idea
	   of using the constant PPID (parent PID) to create unique file names for each session.
	   So, given that we didn't yet find a way to make this work from within MSYS (and likely Cygwin)
	   we are forcing PPID = 0 in all Windows variants unless set via GMT_SESSION_NAME. A corollary of this
	   is that Windows users running many bash windows concurrently should use GMT_SESSION_NAME in their scripts
	   to give unique values to different scripts.  */
```

Would setting a unique `GMT_SESSION_NAME` here in PyGMT result in any issues then? I couldn't run `git blame` on those lines to find out more context.

Reply:
This note (https://docs.generic-mapping-tools.org/dev/begin.html#note-on-unix-shells) is also useful. So, the key point for how GMT modern mode works is having a unique and invariable value so that GMT modules know where to find the session directory and the historical information. The value can be either the constant PPID or a number manually set by `GMT_SESSION_NAME`.

> Would setting a unique `GMT_SESSION_NAME` here in PyGMT result in any issues then? 

All tests still pass, so I believe it has no side effects.

Reply:
Ok, but maybe we should check if `GMT_SESSION_NAME` is set by the user first before overriding it.

Reply:
Users should **NEVER** set `GMT_SESSION_NAME` as a **global** environment variable. Even if they set `GMT_SESSION_NAME`, we should override it in PyGMT.

Here are steps to understand why `GMT_SESSION_NAME` should be overridden. Open a terminal, define the variable:
```
export GMT_SESSION_NAME=123456
```
and run `gmt begin`. Then you'll see the session directory `~/.gmt/sessions/gmt_session.123456/`; all information will be saved in this directory. So, if a global `GMT_SESSION_NAME` is set, processes will write information and output to the same directory, leading to crashes.